tests/system/admin_overlay/test_system_async.py

```diff
@@ -0,0 +1,364 @@
+    )
+
+    # Read all the rows; there shouldn't be that many of them
+    query = read_rows_query.ReadRowsQuery(limit=NUM_ROWS)
+    async for row in await data_client_table.read_rows_stream(query):
+        latest_cell = row[TEST_COLUMMN_FAMILY_NAME, TEST_COLUMN_NAME][0]
+        assert latest_cell.value.decode("utf-8") == value
+
+
+@CrossSync.convert(
+    replace_symbols={
+        "AsyncRestoreTableOperation": "RestoreTableOperation",
+        "AsyncOperation": "Operation",
+    }
+)
+@CrossSync.pytest
+@pytest.mark.parametrize(
+    "second_instance_storage_type,expect_optimize_operation",
+    [
+        (admin_v2.StorageType.HDD, False),
+        (admin_v2.StorageType.SSD, True),
+    ],
+)
+async def test_optimize_restored_table(
+    admin_overlay_project_id,
+    instance_admin_client,
+    table_admin_client,
+    data_client,
+    instances_to_delete,
+    backups_to_delete,
+    second_instance_storage_type,
+    expect_optimize_operation,
+):
+    # Create two instances. We backup a table from the first instance to a new table in the
+    # second instance. This is to test whether or not different scenarios trigger an
+    # optimize_restored_table operation
+    instance_with_backup, table_to_backup = await create_instance(
+        instance_admin_client,
+        table_admin_client,
+        data_client,
+        admin_overlay_project_id,
+        instances_to_delete,
+        admin_v2.StorageType.HDD,
+    )
+
+    instance_to_restore, _ = await create_instance(
+        instance_admin_client,
+        table_admin_client,
+        data_client,
+        admin_overlay_project_id,
+        instances_to_delete,
+        second_instance_storage_type,
+    )
+
+    backup = await create_backup(
+        instance_admin_client,
+        table_admin_client,
+        instance_with_backup,
+        table_to_backup,
+        backups_to_delete,
+    )
+
+    # Restore to other instance
+    restore_operation = await table_admin_client.restore_table(
+        admin_v2.RestoreTableRequest(
+            parent=instance_to_restore.name,
+            table_id=TEST_BACKUP_TABLE_NAME,
+            backup=backup.name,
+        )
+    )
+
+    assert isinstance(restore_operation, admin_v2.AsyncRestoreTableOperation)
+    restored_table = await restore_operation.result()
+
+    optimize_operation = await restore_operation.optimize_restored_table_operation()
+    if expect_optimize_operation:
+        assert isinstance(optimize_operation, api_core_operation.AsyncOperation)
+        await optimize_operation.result()
+    else:
+        assert optimize_operation is None
+
+    # Test that the new table exists
+    assert (
+        restored_table.name
+        == f"{instance_to_restore.name}/tables/{TEST_BACKUP_TABLE_NAME}"
+    )
+    await assert_table_cell_value_equal_to(
+        table_admin_client,
+        data_client,
+        instance_to_restore,
+        restored_table,
+        INITIAL_CELL_VALUE,
+    )
+
+
+@CrossSync.convert
```

Comment:
nit: I think @CrossSync.pytest already does convert automatically, so this should be redundant. but you can double-check